src/app/components/category-products/category-products.component.ts

```diff
@@ -114,22 +135,56 @@ export class CategoryProductsComponent {
     })
   );
 
+  readonly priceFrom: FormControl = new FormControl();
+  readonly priceTo: FormControl = new FormControl();
+
+  constructor(
+    private _activatedRoute: ActivatedRoute,
+    private _categoriesService: CategoriesService,
+    private _productsService: ProductsService,
+    private _router: Router
+  ) {}
+
+  ngOnInit(): void {
```

Comment:
after view init would be better, because you related to elements that are in the view